### PR TITLE
Don't run build and test if only docs changed

### PR DESCRIPTION
**Pull request by lunny**
As title.

But maybe it's affected by https://github.com/actions/runner/issues/2324

### CODE REVIEW

**Comment by lunny** [comment]
> Seems fine with me, but I'm not sure how the "required checks" will work if we skip the required checks.

Ah, yes. Gitea actions will also encounter this like problem.

**Comment by yardenshoham** [comment]
> Seems fine with me, but I'm not sure how the "required checks" will work if we skip the required checks.

Relevant documentation page: https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-protected-branches/troubleshooting-required-status-checks#handling-skipped-but-required-checks

**Review by yardenshoham — APPROVED** [review]

**Review by techknowlogick — APPROVED** [review]

File: .github/workflows/pull-db-tests-docsignore.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+name: db-tests
+
+on:
+  pull_request:
+    paths:
+      - "docs/**"
+      - "*.md"
+
+jobs:
+  test-pgsql:
```
Comment (techknowlogick) [inline .github/workflows/pull-db-tests-docsignore.yml:10]: We need other DB tests to have no build required either

File: .github/workflows/pull-db-tests-docsignore.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - run: echo "No build required"'
```
Comment (techknowlogick) [inline .github/workflows/pull-db-tests-docsignore.yml]: extra `'` at the end of this line

**Review by techknowlogick — CHANGES_REQUESTED** [review]
few minor things

File: .github/workflows/pull-compliance-docsignore.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  pull_request:
+    paths:
+      - "docs/**"
+      - "*.md"
+
+jobs:
+  lint-backend:
+    runs-on: ubuntu-latest
+    steps:
+      - run: echo "No build required"'
```
Comment (techknowlogick) [inline .github/workflows/pull-compliance-docsignore.yml]: here too

File: .github/workflows/pull-docker-dryrun-docsignore.yml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  pull_request:
+    paths:
+      - "docs/**"
+      - "*.md"
+
+jobs:
+  docker-dryrun:
+    runs-on: ubuntu-latest
+    steps:
+      - run: echo "No build required"'
```
Comment (techknowlogick) [inline .github/workflows/pull-docker-dryrun-docsignore.yml]: here too

File: .github/workflows/pull-e2e-tests-docsignore.yml [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+  pull_request:
+    paths:
+      - "docs/**"
+      - "*.md"
+
+jobs:
+  test-e2e:
+    runs-on: ubuntu-latest
+    steps:
+      - run: echo "No build required"'
```
Comment (techknowlogick) [inline .github/workflows/pull-e2e-tests-docsignore.yml]: here too

**Review by techknowlogick — APPROVED** [review]

**Comment by yardenshoham** [comment]
Check https://github.com/go-gitea/gitea/pull/24757

**Comment by silverwind** [comment]
This has some kind of regression where docs build no longer runs for PRs:

PR build was skipped:

https://github.com/go-gitea/gitea/actions/runs/5039932080/jobs/9038389389

It failed on main when landed:

https://github.com/go-gitea/gitea/actions/runs/5040058905/jobs/9038577074

**Comment by silverwind** [comment]
@lunny revert this? It obviously wasn't tested enough, and I'd rather run too many actions than too few.

**Comment by lunny** [comment]
> @lunny revert this? It obviously wasn't tested enough, and I'd rather run too many actions than too few.

We can't test it unless we merge it. I think the problem is at `compliance-docs`, I ever have a discuss with @techknowlogick at https://github.com/go-gitea/gitea/pull/24761#discussion_r1195827815.

**Comment by silverwind** [comment]
Besides it not working, there is more wrong with this PR:

- A clean design should not need workflows that do nothing
- It introduced duplicate workflow names, making the actions overview confusing:

image

So I would still go for a clean revert.

BTW, in theory you can test locally with https://github.com/nektos/act.

**Comment by lunny** [comment]
> Besides it not working, there is more wrong with this PR:
> 
>     * It introduced duplicate workflow names, making the workflow overview confusing.
> 
>     * A clean design should not need workflows that do nothing
> 
> 
> image
> 
> So I would still go for a clean revert.
> 
> BTW, in theory you can test locally with https://github.com/nektos/act.

The change is based on https://github.com/go-gitea/gitea/pull/24530#issuecomment-1536670068

**Comment by silverwind** [comment]
Seems like a huge hack. I'm surprised GH docs even recommend it. Maybe it could still be one workflows and the skip condition could be determined via a shell script instead.

**Pull request merged**